Nama  : Arolina Rinardi

Kelas : Analisis Big Data D

**Tugas Analisi Big data** melakukan proses EDA (Exploratory Data Analysis) dengan mendownload data set dari OULAD
link dataset sebagai berikut.
https://analyse.kmi.open.ac.uk/open_dataset
didalam dataset ini memiliki 7 file csv yang nantinya akan diproses menggunakan EDA.

In [52]:
import pandas as pd


 **Langkah awal**  melakukan import data menggunakan pandas, library ini penting untuk mengolah data yang akan di masukkan nantinya

In [53]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


code diatas merupakan sebuah kode yang digunakan untuk **menghubungkan google drive ke google colab**. Dengan menggunakan kode tersebut maka akan mendapatkan akses dan mengolah file yang ada di google drive secara langsung dari google colab

In [54]:
SA = pd.read_csv ('/content/drive/MyDrive/Analisis Big Data/studentAssessment.csv')
SI = pd.read_csv ('/content/drive/MyDrive/Analisis Big Data/studentInfo.csv')
SR = pd.read_csv ('/content/drive/MyDrive/Analisis Big Data/studentRegistration.csv')
AS = pd.read_csv ('/content/drive/MyDrive/Analisis Big Data/assessments.csv')
CO = pd.read_csv ('/content/drive/MyDrive/Analisis Big Data/courses.csv')
SV = pd.read_csv ('/content/drive/MyDrive/Analisis Big Data/studentVle.csv')
Vl = pd.read_csv ('/content/drive/MyDrive/Analisis Big Data/vle.csv')


**code pd.read_csv** merupakan salah satu fungsi yang disediakan oleh library pandas dalam python digunakan untuk **membaca file dalam format CSV** (COMMA-SEPARATED VALUES) dan mengubahnya menjadi objek DataFrame.

# 1. Student Assassment

In [55]:
SA

id_assessment  id_student  date_submitted  is_banked  score
0                1752       11391              18          0   78.0
1                1752       28400              22          0   70.0
2                1752       31604              17          0   72.0
3                1752       32885              26          0   69.0
4                1752       38053              19          0   79.0
...               ...         ...             ...        ...    ...
173907          37443      527538             227          0   60.0
173908          37443      534672             229          0  100.0
173909          37443      546286             215          0   80.0
173910          37443      546724             230          0  100.0
173911          37443      558486             224          0   80.0

[173912 rows x 5 columns]

**SA** merupakan data dari file csv yang bernama Student Assassment,lalu dipanggil dan memunculkan tampilan data didalam file csv tersebut.

In [103]:
 SA.describe()

id_assessment    id_student  date_submitted      is_banked  \
count  173912.000000  1.739120e+05   173912.000000  173912.000000   
mean    26553.803556  7.051507e+05      116.032942       0.010977   
std      8829.784254  5.523952e+05       71.484148       0.104194   
min      1752.000000  6.516000e+03      -11.000000       0.000000   
25%     15022.000000  5.044290e+05       51.000000       0.000000   
50%     25359.000000  5.852080e+05      116.000000       0.000000   
75%     34883.000000  6.344980e+05      173.000000       0.000000   
max     37443.000000  2.698588e+06      608.000000       1.000000   

               score  
count  173912.000000  
mean       75.799573  
std        18.788755  
min         0.000000  
25%        65.000000  
50%        80.000000  
75%        90.000000  
max       100.000000

**describe()** merupakan sebuah method yang dimiliki oleh python di Pandas digunakan untuk menghitung statistik dasar untuk kolom numerik dan string di DataFrame. Jadi jika kita menginginkan menghitung data statistika dari data student assassment maka dengan perintah memanggil SA.descibe().

In [57]:
SA.dtypes

id_assessment       int64
id_student          int64
date_submitted      int64
is_banked           int64
score             float64
dtype: object

**dtypes** adalah atribut yang tersedia pada objek DataFrame dalam perpustakaan Pandas di Python. Atribut ini digunakan untuk mengembalikan tipe data dari setiap kolom dalam DataFrame.

In [58]:
# Menampilkan metadata dataset
SA.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 173912 entries, 0 to 173911
Data columns (total 5 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   id_assessment   173912 non-null  int64  
 1   id_student      173912 non-null  int64  
 2   date_submitted  173912 non-null  int64  
 3   is_banked       173912 non-null  int64  
 4   score           173739 non-null  float64
dtypes: float64(1), int64(4)
memory usage: 6.6 MB


**Metode info()** di Pandas digunakan untuk mendapatkan ringkasan singkat dari DataFrame, termasuk informasi tentang jumlah baris, jumlah nilai bukan nol di setiap kolom, dan tipe data setiap kolom.

In [59]:
# Mengecek komponen-komponen dataset
print(f"Data Types:\n{SA.dtypes}")
print(f"\nRows and Columns:\n{SA.shape}")
print(f"\nColumn Names:\n{SA.columns}")
print(f"\nNull Values %:\n{SA.apply(lambda x: sum(x.isnull()) / len(SA))}")

Data Types:
id_assessment       int64
id_student          int64
date_submitted      int64
is_banked           int64
score             float64
dtype: object

Rows and Columns:
(173912, 5)

Column Names:
Index(['id_assessment', 'id_student', 'date_submitted', 'is_banked', 'score'], dtype='object')

Null Values %:
id_assessment     0.000000
id_student        0.000000
date_submitted    0.000000
is_banked         0.000000
score             0.000995
dtype: float64


Kode diatas memberikan informasi penting tentang DataFrame SA, termasuk tipe data kolom, ukuran DataFrame, daftar nama kolom, dan persentase nilai null dalam setiap kolom. Informasi ini bermanfaat untuk memahami struktur dan isi DataFrame dan dapat membantu dalam analisis dan pemrosesan data lebih lanjut.

In [60]:
#melihat apakah di data ini memiliki missing value?
SA.isnull().sum()

id_assessment       0
id_student          0
date_submitted      0
is_banked           0
score             173
dtype: int64

Kode SA.isnull().sum()menghitung jumlah nilai yang hilang di setiap kolom dan mengembalikan objek Seri dengan nama kolom sebagai indeks dan jumlah nilai yang hilang sebagai nilainya. jadi dalam data ini memiliki missing value pada fitur score sejumlah 173

In [61]:
# Data Cleaning analisis yang baik
rata = SA['score'].mean()
SA['score'] = SA['score'].fillna(rata)
SA['score'].isna().sum()

0

mengatasi masalah nilai null dengan menggantinya dengan nilai yang masuk akal (rata-rata dalam hal ini). Dengan melakukan hal ini, kita dapat menjaga integritas data dan memastikan bahwa kolom 'score' tidak memiliki nilai null yang dapat mengganggu analisis atau pemrosesan data selanjutnya.

In [62]:
SA.isnull().sum()

id_assessment     0
id_student        0
date_submitted    0
is_banked         0
score             0
dtype: int64

melihat kembali apakah setelah dilakukan menghilangkan missing value pada fitur score.

# 2. Student Info

In [63]:
SI

code_module code_presentation  id_student gender                region  \
0             AAA             2013J       11391      M   East Anglian Region   
1             AAA             2013J       28400      F              Scotland   
2             AAA             2013J       30268      F  North Western Region   
3             AAA             2013J       31604      F     South East Region   
4             AAA             2013J       32885      F  West Midlands Region   
...           ...               ...         ...    ...                   ...   
32588         GGG             2014J     2640965      F                 Wales   
32589         GGG             2014J     2645731      F   East Anglian Region   
32590         GGG             2014J     2648187      F          South Region   
32591         GGG             2014J     2679821      F     South East Region   
32592         GGG             2014J     2684003      F      Yorkshire Region   

           highest_education imd_band age_band  num_of_prev_attempts  \
0           HE Qualification  90-100%     55<=                     0   
1           HE Qualification   20-30%    35-55                     0   
2      A Level or Equivalent   30-40%    35-55                     0   
3      A Level or Equivalent   50-60%    35-55                     0   
4         Lower Than A Level   50-60%     0-35                     0   
...                      ...      ...      ...                   ...   
32588     Lower Than A Level    10-20     0-35                     0   
32589     Lower Than A Level   40-50%    35-55                     0   
32590  A Level or Equivalent   20-30%     0-35                     0   
32591     Lower Than A Level  90-100%    35-55                     0   
32592       HE Qualification   50-60%    35-55                     0   

       studied_credits disability final_result  
0                  240          N         Pass  
1                   60          N         Pass  
2                   60          Y    Withdrawn  
3                   60          N         Pass  
4                   60          N         Pass  
...                ...        ...          ...  
32588               30          N         Fail  
32589               30          N  Distinction  
32590               30          Y         Pass  
32591               30          N    Withdrawn  
32592               30          N  Distinction  

[32593 rows x 12 columns]

menampilkan data pada tabel Student Info maka dengan perintah SI sesuai nama data yang sudah di load ke dalam google drive.

In [64]:
 SI.describe()

id_student  num_of_prev_attempts  studied_credits
count  3.259300e+04          32593.000000     32593.000000
mean   7.066877e+05              0.163225        79.758691
std    5.491673e+05              0.479758        41.071900
min    3.733000e+03              0.000000        30.000000
25%    5.085730e+05              0.000000        60.000000
50%    5.903100e+05              0.000000        60.000000
75%    6.444530e+05              0.000000       120.000000
max    2.716795e+06              6.000000       655.000000

**describe()** merupakan sebuah method yang dimiliki oleh python di Pandas digunakan untuk menghitung statistik dasar untuk kolom numerik dan string di DataFrame. Jadi jika kita menginginkan menghitung data statistika dari data student assassment maka dengan perintah memanggil SA.descibe().

menampikkan hitungan statistika pada data Student Info menggunakan Describe()

In [65]:
# Menampilkan metadata dataset
SI.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32593 entries, 0 to 32592
Data columns (total 12 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   code_module           32593 non-null  object
 1   code_presentation     32593 non-null  object
 2   id_student            32593 non-null  int64 
 3   gender                32593 non-null  object
 4   region                32593 non-null  object
 5   highest_education     32593 non-null  object
 6   imd_band              31482 non-null  object
 7   age_band              32593 non-null  object
 8   num_of_prev_attempts  32593 non-null  int64 
 9   studied_credits       32593 non-null  int64 
 10  disability            32593 non-null  object
 11  final_result          32593 non-null  object
dtypes: int64(3), object(9)
memory usage: 3.0+ MB


**Metode info()** di Pandas digunakan untuk mendapatkan ringkasan singkat dari DataFrame, termasuk informasi tentang jumlah baris, jumlah nilai bukan nol di setiap kolom, dan tipe data setiap kolom.

In [66]:
# Mengecek komponen-komponen dataset
print(f"Data Types:\n{SI.dtypes}")
print(f"\nRows and Columns:\n{SI.shape}")
print(f"\nColumn Names:\n{SI.columns}")
print(f"\nNull Values %:\n{SI.apply(lambda x: sum(x.isnull()) / len(SI))}")

Data Types:
code_module             object
code_presentation       object
id_student               int64
gender                  object
region                  object
highest_education       object
imd_band                object
age_band                object
num_of_prev_attempts     int64
studied_credits          int64
disability              object
final_result            object
dtype: object

Rows and Columns:
(32593, 12)

Column Names:
Index(['code_module', 'code_presentation', 'id_student', 'gender', 'region',
       'highest_education', 'imd_band', 'age_band', 'num_of_prev_attempts',
       'studied_credits', 'disability', 'final_result'],
      dtype='object')

Null Values %:
code_module             0.000000
code_presentation       0.000000
id_student              0.000000
gender                  0.000000
region                  0.000000
highest_education       0.000000
imd_band                0.034087
age_band                0.000000
num_of_prev_attempts    0.000000
studied_cre

Kode diatas memberikan informasi penting tentang DataFrame SI, termasuk tipe data kolom, ukuran DataFrame, daftar nama kolom, dan persentase nilai null dalam setiap kolom. Informasi ini bermanfaat untuk memahami struktur dan isi DataFrame dan dapat membantu dalam analisis dan pemrosesan data lebih lanjut.

In [67]:
SI.isnull().sum()

code_module                0
code_presentation          0
id_student                 0
gender                     0
region                     0
highest_education          0
imd_band                1111
age_band                   0
num_of_prev_attempts       0
studied_credits            0
disability                 0
final_result               0
dtype: int64

memeriksa apakah data studnet info memiliki missing value. jadi pada data studnet info memmiiki missing value pada fitur imd_band sebanyak 1111

In [68]:
SI['imd_band'].fillna('unknown', inplace=True)

Kode SI['imd_band'].fillna('unknown', inplace=True)digunakan untuk mengisi nilai yang hilang (nilai null) pada kolom 'imd_band' DataFrame SI dengan nilai 'unknown'.

In [69]:
SI.isnull().sum()

code_module             0
code_presentation       0
id_student              0
gender                  0
region                  0
highest_education       0
imd_band                0
age_band                0
num_of_prev_attempts    0
studied_credits         0
disability              0
final_result            0
dtype: int64

melihat dan memastikan tidak ada missing value

# 3. Student Registration

In [70]:
SR

code_module code_presentation  id_student  date_registration  \
0             AAA             2013J       11391             -159.0   
1             AAA             2013J       28400              -53.0   
2             AAA             2013J       30268              -92.0   
3             AAA             2013J       31604              -52.0   
4             AAA             2013J       32885             -176.0   
...           ...               ...         ...                ...   
32588         GGG             2014J     2640965               -4.0   
32589         GGG             2014J     2645731              -23.0   
32590         GGG             2014J     2648187             -129.0   
32591         GGG             2014J     2679821              -49.0   
32592         GGG             2014J     2684003              -28.0   

       date_unregistration  
0                      NaN  
1                      NaN  
2                     12.0  
3                      NaN  
4                      NaN  
...                    ...  
32588                  NaN  
32589                  NaN  
32590                  NaN  
32591                101.0  
32592                  NaN  

[32593 rows x 5 columns]

membaca atau melihat tabel pada Student Registration

In [71]:
 SR.describe()

id_student  date_registration  date_unregistration
count  3.259300e+04       32548.000000         10072.000000
mean   7.066877e+05         -69.411300            49.757645
std    5.491673e+05          49.260522            82.460890
min    3.733000e+03        -322.000000          -365.000000
25%    5.085730e+05        -100.000000            -2.000000
50%    5.903100e+05         -57.000000            27.000000
75%    6.444530e+05         -29.000000           109.000000
max    2.716795e+06         167.000000           444.000000

melakukan perhitungan statistika pada tabel Student Registration

In [72]:
# Menampilkan metadata dataset
SR.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32593 entries, 0 to 32592
Data columns (total 5 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   code_module          32593 non-null  object 
 1   code_presentation    32593 non-null  object 
 2   id_student           32593 non-null  int64  
 3   date_registration    32548 non-null  float64
 4   date_unregistration  10072 non-null  float64
dtypes: float64(2), int64(1), object(2)
memory usage: 1.2+ MB


Metode info() di Pandas digunakan untuk mendapatkan ringkasan singkat dari DataFrame, termasuk informasi tentang jumlah baris, jumlah nilai bukan nol di setiap kolom, dan tipe data setiap kolom.

In [73]:
# Mengecek komponen-komponen dataset
print(f"Data Types:\n{SR.dtypes}")
print(f"\nRows and Columns:\n{SR.shape}")
print(f"\nColumn Names:\n{SR.columns}")
print(f"\nNull Values %:\n{SR.apply(lambda x: sum(x.isnull()) / len(SR))}")

Data Types:
code_module             object
code_presentation       object
id_student               int64
date_registration      float64
date_unregistration    float64
dtype: object

Rows and Columns:
(32593, 5)

Column Names:
Index(['code_module', 'code_presentation', 'id_student', 'date_registration',
       'date_unregistration'],
      dtype='object')

Null Values %:
code_module            0.000000
code_presentation      0.000000
id_student             0.000000
date_registration      0.001381
date_unregistration    0.690977
dtype: float64


Kode diatas memberikan informasi penting tentang DataFrame SR, termasuk tipe data kolom, ukuran DataFrame, daftar nama kolom, dan persentase nilai null dalam setiap kolom. Informasi ini bermanfaat untuk memahami struktur dan isi DataFrame dan dapat membantu dalam analisis dan pemrosesan data lebih lanjut.

In [74]:
SR.isnull().sum()

code_module                0
code_presentation          0
id_student                 0
date_registration         45
date_unregistration    22521
dtype: int64

melihat missing value ada 2 fitur yaitu date_registration dan date_unregistration  

In [75]:
SR['date_registration'].fillna('unknown', inplace=True)

menghilangkan missing value

In [76]:
SR['date_unregistration'].fillna('unknown', inplace=True)

In [77]:
SR.isnull().sum()

code_module            0
code_presentation      0
id_student             0
date_registration      0
date_unregistration    0
dtype: int64

melihat kembali memastikan tidak ada missing value

# 4. Assessments

In [78]:
AS

code_module code_presentation  id_assessment assessment_type   date  \
0           AAA             2013J           1752             TMA   19.0   
1           AAA             2013J           1753             TMA   54.0   
2           AAA             2013J           1754             TMA  117.0   
3           AAA             2013J           1755             TMA  166.0   
4           AAA             2013J           1756             TMA  215.0   
..          ...               ...            ...             ...    ...   
201         GGG             2014J          37443             CMA  229.0   
202         GGG             2014J          37435             TMA   61.0   
203         GGG             2014J          37436             TMA  124.0   
204         GGG             2014J          37437             TMA  173.0   
205         GGG             2014J          37444            Exam  229.0   

     weight  
0      10.0  
1      20.0  
2      20.0  
3      20.0  
4      30.0  
..      ...  
201     0.0  
202     0.0  
203     0.0  
204     0.0  
205   100.0  

[206 rows x 6 columns]

memanggil tabel assessment

In [79]:
 AS.describe()

id_assessment        date      weight
count     206.000000  195.000000  206.000000
mean    26473.975728  145.005128   20.873786
std     10098.625521   76.001119   30.384224
min      1752.000000   12.000000    0.000000
25%     15023.250000   71.000000    0.000000
50%     25364.500000  152.000000   12.500000
75%     34891.750000  222.000000   24.250000
max     40088.000000  261.000000  100.000000

melakukan perhitungan statistika pada tabel Assessment

In [80]:
# Menampilkan metadata dataset
AS.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 206 entries, 0 to 205
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   code_module        206 non-null    object 
 1   code_presentation  206 non-null    object 
 2   id_assessment      206 non-null    int64  
 3   assessment_type    206 non-null    object 
 4   date               195 non-null    float64
 5   weight             206 non-null    float64
dtypes: float64(2), int64(1), object(3)
memory usage: 9.8+ KB


In [81]:
# Mengecek komponen-komponen dataset
print(f"Data Types:\n{AS.dtypes}")
print(f"\nRows and Columns:\n{AS.shape}")
print(f"\nColumn Names:\n{AS.columns}")
print(f"\nNull Values %:\n{AS.apply(lambda x: sum(x.isnull()) / len(AS))}")

Data Types:
code_module           object
code_presentation     object
id_assessment          int64
assessment_type       object
date                 float64
weight               float64
dtype: object

Rows and Columns:
(206, 6)

Column Names:
Index(['code_module', 'code_presentation', 'id_assessment', 'assessment_type',
       'date', 'weight'],
      dtype='object')

Null Values %:
code_module          0.000000
code_presentation    0.000000
id_assessment        0.000000
assessment_type      0.000000
date                 0.053398
weight               0.000000
dtype: float64


melihat detail pada tabel

In [82]:
AS.isnull().sum()

code_module           0
code_presentation     0
id_assessment         0
assessment_type       0
date                 11
weight                0
dtype: int64

memiliki missing value pada fitur date sebanyak 11

In [83]:
AS['date'].fillna('unknown', inplace=True)

melakukan pembersihan missing value

In [84]:
AS.isnull().sum()

code_module          0
code_presentation    0
id_assessment        0
assessment_type      0
date                 0
weight               0
dtype: int64

melihat kembali missing value

# 5. Courses

In [85]:
CO

code_module code_presentation  module_presentation_length
0          AAA             2013J                         268
1          AAA             2014J                         269
2          BBB             2013J                         268
3          BBB             2014J                         262
4          BBB             2013B                         240
5          BBB             2014B                         234
6          CCC             2014J                         269
7          CCC             2014B                         241
8          DDD             2013J                         261
9          DDD             2014J                         262
10         DDD             2013B                         240
11         DDD             2014B                         241
12         EEE             2013J                         268
13         EEE             2014J                         269
14         EEE             2014B                         241
15         FFF             2013J                         268
16         FFF             2014J                         269
17         FFF             2013B                         240
18         FFF             2014B                         241
19         GGG             2013J                         261
20         GGG             2014J                         269
21         GGG             2014B                         241

memanggil tabel courses

In [86]:
 CO.describe()

module_presentation_length
count                   22.000000
mean                   255.545455
std                     13.654677
min                    234.000000
25%                    241.000000
50%                    261.500000
75%                    268.000000
max                    269.000000

menghitung statistika pada tabel courses

In [87]:
# Menampilkan metadata dataset
CO.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 3 columns):
 #   Column                      Non-Null Count  Dtype 
---  ------                      --------------  ----- 
 0   code_module                 22 non-null     object
 1   code_presentation           22 non-null     object
 2   module_presentation_length  22 non-null     int64 
dtypes: int64(1), object(2)
memory usage: 656.0+ bytes


In [88]:
# Mengecek komponen-komponen dataset
print(f"Data Types:\n{CO.dtypes}")
print(f"\nRows and Columns:\n{CO.shape}")
print(f"\nColumn Names:\n{CO.columns}")
print(f"\nNull Values %:\n{CO.apply(lambda x: sum(x.isnull()) / len(CO))}")

Data Types:
code_module                   object
code_presentation             object
module_presentation_length     int64
dtype: object

Rows and Columns:
(22, 3)

Column Names:
Index(['code_module', 'code_presentation', 'module_presentation_length'], dtype='object')

Null Values %:
code_module                   0.0
code_presentation             0.0
module_presentation_length    0.0
dtype: float64


In [89]:
CO.isnull().sum()

code_module                   0
code_presentation             0
module_presentation_length    0
dtype: int64

# 6. Student Vle

In [90]:
SV

code_module code_presentation  id_student  id_site  date  sum_click
0                AAA             2013J       28400   546652   -10          4
1                AAA             2013J       28400   546652   -10          1
2                AAA             2013J       28400   546652   -10          1
3                AAA             2013J       28400   546614   -10         11
4                AAA             2013J       28400   546714   -10          1
...              ...               ...         ...      ...   ...        ...
10655275         GGG             2014J      675811   896943   269          3
10655276         GGG             2014J      675578   896943   269          1
10655277         GGG             2014J      654064   896943   269          3
10655278         GGG             2014J      654064   896939   269          1
10655279         GGG             2014J      654064   896939   269          1

[10655280 rows x 6 columns]

membaca atau melihat tabel Student Vle

In [91]:
 SV.describe()

id_student       id_site          date     sum_click
count  1.065528e+07  1.065528e+07  1.065528e+07  1.065528e+07
mean   7.333336e+05  7.383234e+05  9.517400e+01  3.716946e+00
std    5.827060e+05  1.312196e+05  7.607130e+01  8.849047e+00
min    6.516000e+03  5.267210e+05 -2.500000e+01  1.000000e+00
25%    5.077430e+05  6.735190e+05  2.500000e+01  1.000000e+00
50%    5.882360e+05  7.300690e+05  8.600000e+01  2.000000e+00
75%    6.464840e+05  8.770300e+05  1.560000e+02  3.000000e+00
max    2.698588e+06  1.049562e+06  2.690000e+02  6.977000e+03

mengitung statistika pada tabel Student Vle

In [92]:
# Menampilkan metadata dataset
SV.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10655280 entries, 0 to 10655279
Data columns (total 6 columns):
 #   Column             Dtype 
---  ------             ----- 
 0   code_module        object
 1   code_presentation  object
 2   id_student         int64 
 3   id_site            int64 
 4   date               int64 
 5   sum_click          int64 
dtypes: int64(4), object(2)
memory usage: 487.8+ MB


In [93]:
# Mengecek komponen-komponen dataset
print(f"Data Types:\n{SV.dtypes}")
print(f"\nRows and Columns:\n{SV.shape}")
print(f"\nColumn Names:\n{SV.columns}")
print(f"\nNull Values %:\n{SV.apply(lambda x: sum(x.isnull()) / len(SV))}")

Data Types:
code_module          object
code_presentation    object
id_student            int64
id_site               int64
date                  int64
sum_click             int64
dtype: object

Rows and Columns:
(10655280, 6)

Column Names:
Index(['code_module', 'code_presentation', 'id_student', 'id_site', 'date',
       'sum_click'],
      dtype='object')

Null Values %:
code_module          0.0
code_presentation    0.0
id_student           0.0
id_site              0.0
date                 0.0
sum_click            0.0
dtype: float64


melihat detail tabel student vle

In [94]:
SV.isnull().sum()

code_module          0
code_presentation    0
id_student           0
id_site              0
date                 0
sum_click            0
dtype: int64

tidak memiliki missing value pada tabel student vle

# 7. Vle

In [95]:
Vl

id_site code_module code_presentation activity_type  week_from  week_to
0      546943         AAA             2013J      resource        NaN      NaN
1      546712         AAA             2013J     oucontent        NaN      NaN
2      546998         AAA             2013J      resource        NaN      NaN
3      546888         AAA             2013J           url        NaN      NaN
4      547035         AAA             2013J      resource        NaN      NaN
...       ...         ...               ...           ...        ...      ...
6359   897063         GGG             2014J      resource        NaN      NaN
6360   897109         GGG             2014J      resource        NaN      NaN
6361   896965         GGG             2014J     oucontent        NaN      NaN
6362   897060         GGG             2014J      resource        NaN      NaN
6363   897100         GGG             2014J      resource        NaN      NaN

[6364 rows x 6 columns]

membaca tabel vle

In [96]:
 Vl.describe()

id_site    week_from      week_to
count  6.364000e+03  1121.000000  1121.000000
mean   7.260991e+05    15.204282    15.214987
std    1.283151e+05     8.792865     8.779806
min    5.267210e+05     0.000000     0.000000
25%    6.615928e+05     8.000000     8.000000
50%    7.300965e+05    15.000000    15.000000
75%    8.140162e+05    22.000000    22.000000
max    1.077905e+06    29.000000    29.000000

menghitung statistika pada tabel vle

In [97]:
# Menampilkan metadata dataset
Vl.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6364 entries, 0 to 6363
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id_site            6364 non-null   int64  
 1   code_module        6364 non-null   object 
 2   code_presentation  6364 non-null   object 
 3   activity_type      6364 non-null   object 
 4   week_from          1121 non-null   float64
 5   week_to            1121 non-null   float64
dtypes: float64(2), int64(1), object(3)
memory usage: 298.4+ KB


In [98]:
# Mengecek komponen-komponen dataset
print(f"Data Types:\n{Vl.dtypes}")
print(f"\nRows and Columns:\n{Vl.shape}")
print(f"\nColumn Names:\n{Vl.columns}")
print(f"\nNull Values %:\n{Vl.apply(lambda x: sum(x.isnull()) / len(Vl))}")

Data Types:
id_site                int64
code_module           object
code_presentation     object
activity_type         object
week_from            float64
week_to              float64
dtype: object

Rows and Columns:
(6364, 6)

Column Names:
Index(['id_site', 'code_module', 'code_presentation', 'activity_type',
       'week_from', 'week_to'],
      dtype='object')

Null Values %:
id_site              0.000000
code_module          0.000000
code_presentation    0.000000
activity_type        0.000000
week_from            0.823853
week_to              0.823853
dtype: float64


melihat detail singkat tentang tabel vle

In [99]:
Vl.isnull().sum()

id_site                 0
code_module             0
code_presentation       0
activity_type           0
week_from            5243
week_to              5243
dtype: int64

melihat missing value pada week_from sebanyak 5243 dan week_to sebanyak 5243

In [100]:
Vl['week_from'].fillna('unknown', inplace=True)

membersihkan missing value

In [101]:
Vl['week_to'].fillna('unknown', inplace=True)

In [102]:
Vl.isnull().sum()

id_site              0
code_module          0
code_presentation    0
activity_type        0
week_from            0
week_to              0
dtype: int64

memastikan tidak ada missing value